docs/crocks/Pred.md

```diff
@@ -0,0 +1,57 @@
+# Pred
+
+`Pred a`
+
+--
+
+--
+
+```js
+--
+```
+
+`Pred` exposes the following functions on the constructor and instance:
+
+| Constructor | Instance |
+|:---|:---|
+| [`empty`](#empty), [`type`](#type) | [`concat`](#concat), [`contramap`](#contramap), [`empty`](#empty),  [`inspect`](#inspect), [`runWith`](#runWith), [`type`](#type), [`value`](#value) |
+
+## Constructor
+
+### empty
+
+`Pred m => () ~> m ()`
+
+### type
+
+`() -> String`
+
+## Instance
+
+### concat
+
+`Pred m => m a ~> a -> m a`
+
+### contramap
+
+`Pred m => m a ~> a -> m ()`
```

Comment:
Now this one I KNEW was going to trip you up, because the signature is strange a heck.

It looks like this:
`Pred m => m a ~> (b -> a) -> m b`

So contravarience is use for things like: I have a function `(Number -> Boolean)`, but I do not have `Numbers`, say I have an `Object` that has a `Number` on it. I need to adapt my Pred somehow: 
```js

// pred : Pred Number
const pred = Pred(x => x > 10)

// objPred : Pred { 'num' }
const objPred =
  pred
    .contramap(compose(option(0), prop('num'))

console.log(pred.runWith(3)) // false
console.log(pred.runWith(30)) // true

console.log(objPred.runWith({ num: 8 })) // false
console.log(objPred.runWith({})) // false
console.log(objPred.runWith({ num: 11 })) // true
```